examples/exex/rollup/src/main.rs

```diff
@@ -0,0 +1,558 @@
+    EthEvmConfig::fill_cfg_and_block_env(
+        &mut cfg,
+        evm.block_mut(),
+        &CHAIN_SPEC,
+        &header,
+        U256::ZERO,
+    );
+    *evm.cfg_mut() = cfg.cfg_env;
+
+    let mut receipts = Vec::with_capacity(transactions.len());
+    let mut executed_txs = Vec::with_capacity(transactions.len());
+    let mut results = Vec::with_capacity(transactions.len());
+    if !transactions.is_empty() {
+        let mut cumulative_gas_used = 0;
+        for (transaction, sender) in transactions {
+            // The sum of the transaction’s gas limit, Tg, and the gas utilized in this block prior,
+            // must be no greater than the block’s gasLimit.
+            let block_available_gas = header.gas_limit - cumulative_gas_used;
+            if transaction.gas_limit() > block_available_gas {
+                // TODO(alexey): what to do here?
+                return Err(BlockValidationError::TransactionGasLimitMoreThanAvailableBlockGas {
+                    transaction_gas_limit: transaction.gas_limit(),
+                    block_available_gas,
+                }
+                .into())
+            }
+            // Execute transaction.
+            // Fill revm structure.
+            fill_tx_env(evm.tx_mut(), &transaction, sender);
+
+            let ResultAndState { result, state } = match evm.transact() {
+                Ok(result) => result,
+                Err(err) => {
+                    match err {
+                        EVMError::Transaction(err) => {
+                            // if the transaction is invalid, we can skip it
+                            debug!(%err, ?transaction, "Skipping invalid transaction");
+                            continue
+                        }
+                        err => {
+                            // this is an error that we should treat as fatal for this attempt
+                            return Err(PayloadBuilderError::EvmExecutionError(err).into())
+                        }
+                    }
+                }
+            };
+
+            debug!(?transaction, ?result, ?state, "Executed transaction");
+
+            evm.db_mut().commit(state);
+
+            // append gas used
+            cumulative_gas_used += result.gas_used();
+
+            // Push transaction changeset and calculate header bloom filter for receipt.
+            #[allow(clippy::needless_update)] // side-effect of optimism fields
+            receipts.push(Receipt {
+                tx_type: transaction.tx_type(),
+                success: result.is_success(),
+                cumulative_gas_used,
+                logs: result.logs().iter().cloned().map(Into::into).collect(),
+                ..Default::default()
+            });
+
+            // append transaction to the list of executed transactions
+            executed_txs.push(transaction);
+            results.push(result);
+        }
+
+        evm.db_mut().merge_transitions(BundleRetention::Reverts);
+    }
```

Comment:
why ::Reverts and not PlainState here?
tbh don't really know the difference

Reply:
otherwise reverts aren't persisted, but we need them for insertion into the database to be able to revert on re-orgs